### PR TITLE
Assignment1 by nishtha

### PR DESCRIPTION
**Pull request by Nishthakumari**
assignment1bynishtha

### CODE REVIEW

File: 96284658_nishthakumari/assignment1.cu [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    // Allocating device memory
+    cudaMalloc((void**)&d_a,  Array_Bytes);
+    cudaMalloc((void**)&d_b,  Array_Bytes);
+    cudaMalloc((void**)&d_c,  Array_Bytes);
+
+    // Copying input operands from host to device
+    // For the GPU to perform any operation, the data has to be present in the GPU memory
+    cudaMemcpy(d_a, &h_a,  Array_Bytes, cudaMemcpyHostToDevice);
+    cudaMemcpy(d_b, &h_b,  Array_Bytes, cudaMemcpyHostToDevice);
```
Comment (niranjansy) [inline 96284658_nishthakumari/assignment1.cu:48]: In the cudaMemcpy statement, while mentioning the host pointer, the & is not required before h_a and h_b. Note that h_a and h_b are array names, and an array name is itself a pointer to the first element in the array. So &h_a makes it a double pointer which is incorrect.

File: 96284658_nishthakumari/assignment1.cu [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    // Launching kernel with 1 block and 1 thread
+    // The host launches the kernel on the device
+   
+    Add<<<ceil(1.0*N/1024), 1024>>>(d_a, d_b, d_c,N);
+
+    // Copying the result from device to host
+    cudaMemcpy(&h_c, d_c, Array_Bytes, cudaMemcpyDeviceToHost);
```
Comment (niranjansy) [inline 96284658_nishthakumari/assignment1.cu:56]: Once again, & is not required before h_c.

File: 96284658_nishthakumari/assignment1.cu (hunk 1 above)
Comment (Nishthakumari) [reply 96284658_nishthakumari/assignment1.cu:48]: okay. but I tried running the program and it still worked.

**Review by niranjansy — CHANGES_REQUESTED** [review]
Hi! Please make the mentioned changes to the code. Also, note that the pull request had to be sent to the cuda_c++_smp branch of the repo and not to the master branch. So please make the changes to the code, push the changes and edit the PR to send it to the cuda_c++_smp branch.

**Pull request closed without merging**